framework/src/media/utils/MediaUtils.cpp

```diff
@@ -68,10 +72,257 @@ audio_type_t getAudioTypeFromPath(std::string datapath)
+{
+	*pcmFormat = AUDIO_FORMAT_TYPE_S16_LE;
+	
+	unsigned char bit;
+	unsigned int mpeg_version;	
+	bit = header[1];
+	bit >>= 3;
+	bit &= (unsigned char)3;
+
+	switch (bit) {
+	case 0 :
+		mpeg_version = 2;
+		break;
+	case 2 :
+		mpeg_version = 1;
+		break;
+	case 3 :
+		mpeg_version = 0;
+		break;
+	default :
+		medvdbg("mpeg version is wrong\n");
+		return false;
+	}
+
+	bit = header[2];
+	bit >>= 2;
+	bit &= 3;
+	
+	switch (bit) {
+	case 0 :
+		*sample_rate = 44100;
+		break;
+	case 1 :
+		*sample_rate = 48000;
+		break;
+	case 2 :
+		*sample_rate = 32000;
+		break;
+	default :
+		medvdbg("sample rate is wrong\n");
+		return false;
+	}
+
+	*sample_rate >>= mpeg_version;
+
+	bit = header[3];
+	bit >>= 6;
+	if (bit <= 2) {
+		*channel = 2;
+	} else {
+		*channel = 1;
+	}
+
+	return true;
+}
+
+bool aac_header_parsing(unsigned char *header, unsigned int *channel, unsigned int *sample_rate, audio_format_type_t *pcmFormat)
+{
+	*pcmFormat = AUDIO_FORMAT_TYPE_S16_LE;
+	
+	unsigned char bit;
+	bit = header[2];
+	bit >>= 2;
+	bit &= 15;
+	
+	switch (bit) {
+	case 0 :
+		*sample_rate = 96000;
+		break;
+	case 1 :
+		*sample_rate = 88200;
+		break;
+	case 2 :
+		*sample_rate = 64000;
+		break;
+	case 3 :
+		*sample_rate = 48000;
+		break;
+	case 4 :
+		*sample_rate = 44100;
+		break;
+	case 5 :
+		*sample_rate = 32000;
+		break;
+	case 6 :
+		*sample_rate = 24000;
+		break;
+	case 7 :
+		*sample_rate = 22050;
+		break;
+	case 8 :
+		*sample_rate = 16000;
+		break;
+	case 9 :
+		*sample_rate = 12000;
+		break;
+	case 10 :
+		*sample_rate = 11025;
+		break;
+	case 11 :
+		*sample_rate = 8000;
+		break;
+	case 12 :
+		*sample_rate = 7350;
+		break;
+	default :
+		medvdbg("sample rate is wrong\n");
+		return false;
+	}
+	bit = header[2];
+	bit <<= 2;
+	bit |= (header[3]>>6);
+	bit &= 7;
+	switch (bit) {
+	case 0 :
+		medvdbg("channel is wrong\n");
+		return false;
+	case 1 :
+		*channel = 1;
+		break;
+	case 2 :
+		*channel = 2;
+		break;
+	case 3 :
+		*channel = 3;
+		break;
+	case 4 :
+		*channel = 4;
+		break;
+	case 5 :
+		*channel = 5;
+		break;
+	case 6 :
+		*channel = 6;
+		break;
+	case 7 :
+		*channel = 8;
+		break;
+	default :
+		medvdbg("channel is wrong\n");
+		return false;
+	}
+
+	return true;
+}
+
+bool wave_header_parsing(unsigned char *header, unsigned int *channel, unsigned int *sample_rate, audio_format_type_t *pcmFormat) //header 44byte
+{
+	*channel = header[22];
+
+	*sample_rate = header[25];
+	*sample_rate <<= 8;
+	*sample_rate |= header[24];
+
+	unsigned short bit_per_sample = 0;
+
+	bit_per_sample |= header[35];
+	bit_per_sample <<= 8;
+	bit_per_sample |= header[34];
+
+	switch (bit_per_sample) {
+	case 8 :
+		*pcmFormat = AUDIO_FORMAT_TYPE_S8;
+		break;
+	case 16 :
+		*pcmFormat = AUDIO_FORMAT_TYPE_S16_LE;
+		break;
+	case 32 :
+		*pcmFormat = AUDIO_FORMAT_TYPE_S32_LE;
+		break;
+	default :
+		medvdbg("bit per sample is wrong.\n");
```

Comment:
Default is error case, right?
meddbg("Not Supported Format\n bit/sample : %d", bit_per_sample);